File: RevEachElemLinkedList/Reverse.js
var NODE = /** @class */ (function () {
    function NODE() {
        this.data = 0;
        this.next = null;
    }
    return NODE;
}());
var SinglyLL = /** @class */ (function () {
    function SinglyLL() {
        this.Head = null;
    }
    SinglyLL.prototype.InsertFirst = function (value) {
        var newn = new NODE();
        newn.data = value;
        newn.next = null;
        if (this.Head == null) {
            this.Head = newn;
        }
        else {
            newn.next = this.Head;
            this.Head = newn;
        }
    };
    SinglyLL.prototype.InsertLast = function (value) {
        var newn = new NODE();
        newn.data = value;
        newn.next = null;
        var temp = this.Head;
        if (this.Head == null) {
            this.Head = newn;
        }
        else {
            while (temp.next != null) {
                temp = temp.next;
            }
            temp.next = newn;
        }
    };
    SinglyLL.prototype.Display = function () {
        var Head = this.Head;
        process.stdout.write("Elements of Linked List are : ");
        while (Head != null) {
            process.stdout.write("|" + Head.data + "|->");
            Head = Head.next;
        }
        console.log();
    };
    SinglyLL.prototype.Count = function () {
        var Head = this.Head;
        var icnt = 0;
        while (Head != null) {
            icnt++;
            Head = Head.next;
        }
        return icnt;
    };
    SinglyLL.prototype.Reverse = function () {
        var Head = this.Head;
        var irev = 0;
        var Digit = 0;
        var No = 0;
        process.stdout.write("Ouput After Reversing Each Elements is : ");
        while (Head != null) {
            No = Head.data;
            while (No != 0) {
                Digit = No % 10;
                irev = (irev * 10) + Digit;
                No = Math.floor(No / 10);
            }
            process.stdout.write("|" + irev + "|->");
            irev = 0;
            Head = Head.next;
        }
    };
    return SinglyLL;
}());
function main() {
    var iret = 0;
    console.log("Jay Ganesh....");
    var sobj = new SinglyLL();
    sobj.InsertFirst(51);
    sobj.InsertFirst(11);
    sobj.InsertFirst(10);
    sobj.InsertLast(101);
    sobj.InsertLast(121);
    sobj.InsertLast(1201);
    sobj.Display();
    iret = sobj.Count();
    console.log("Count of elements From Linked List is : ", iret);
    sobj.Reverse();
}
main();

File: RevEachElemLinkedList/Reverse.py
class Node :
    data = 0;
    next = None;

    def __init__(self) :
        self.data = 0;
        self.next = None;

class SinglyLL : 
    Head = None;

    def __init__(self) :
        self.Head = None;
    
    def InsertFirst(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;

        if(self.Head == None) :
            self.Head = newn;
        else :
            newn.next = self.Head;
            self.Head = newn;

    def InsertLast(self,value) :
        newn = Node();
        newn.data = value;
        newn.next = None;
        temp = self.Head;

        if(self.Head == None) :
            self.Head = newn;
        else :
            while(temp.next != None) :
                temp = temp.next;
            temp.next = newn;

    def Display(self) :
        Head = self.Head;

        print("Elements of linked List are as Follows : ");
        while(Head != None) :
            print("|%d|->"%Head.data,end = "");
            Head = Head.next;
        print();

    def Count(self) :
        icnt = 0;
        Head = self.Head;

        while(Head != None) :
            icnt = icnt + 1;
            Head = Head.next;
        
        return icnt;

    def Reverse(self) :
        Head = self.Head;
        irev = 0;
        Digit = 0;
        No = 0;

        print("Output After Reversing Each Elements of Linked list is : ",end = "");
        while(Head != None) :
            No = Head.data;

            while(No != 0) :
                Digit = No % 10;
                irev = (irev * 10) + Digit;
                No = No // 10;
            
            print("|%d|->"%irev,end = "");
            irev = 0;
            Head = Head.next;
    
def main() :
    print("Jay Ganesh....");
    iret = 0;

    sobj = SinglyLL();
    
    sobj.InsertFirst(51);
    sobj.InsertFirst(11);
    sobj.InsertFirst(10);
    sobj.InsertLast(101);
    sobj.InsertLast(1212);
    sobj.InsertLast(999983);

    sobj.Display();
    
    iret = sobj.Count();    
    print("Count of Elements of Linked list is : ",iret);

    sobj.Reverse();

if __name__ == "__main__" :
    main();